x/leverage/keeper/oracle.go

```diff
@@ -13,17 +13,15 @@ import (
 // must be the base denomination, e.g. uumee. The x/oracle module must know of
 // the base and display/symbol denominations for each exchange pair. E.g. it must
 // know about the UMEE/USD exchange rate along with the uumee base denomination
-// and the exponent.
-// This function will only return positive exchange rates or errors, unless a
-// token is blacklisted, in which case it will return zero.
+// and the exponent. When error is nil, price is guaranteed to be positive.
 func (k Keeper) TokenPrice(ctx sdk.Context, denom string) (sdk.Dec, error) {
 	t, err := k.GetTokenSettings(ctx, denom)
 	if err != nil {
 		return sdk.ZeroDec(), err
 	}
 
 	if t.Blacklist {
-		return sdk.ZeroDec(), nil
+		return sdk.ZeroDec(), types.ErrBlacklisted
```

Comment:
I was able to restore the strict nonzero behavior of TokenPrice as a result of other functions filtering blacklisted tokens before calling TokenPrice.